rust/lance-index/src/vector/pq.rs

```diff
@@ -130,15 +139,30 @@ impl ProductQuantizer {
                             num_sub_vectors,
                             sub_idx,
                         );
-                        compute_partition(centroids, sub_vector, distance_type).map(|v| v as u8)
+                        compute_partition(centroids, sub_vector, distance_type).unwrap() as u8
                     })
-                    .collect::<Vec<_>>()
+                    .collect::<Vec<_>>();
+                if num_bits == 4 {
```

Comment:
This feels like a potential bottleneck. Let's have a ticket tracking these and maybe turn it in to generics?